src/misc/programs.cpp

```diff
@@ -53,33 +54,27 @@ constexpr uint8_t exe_block[] = {
 
 // COM data constants
 constexpr int callback_pos = 12;
-constexpr int com_data_size = 32;
 
 // Persistent program containers
-using com_data_t = std::unique_ptr<uint8_t[]>;
-static std::vector<com_data_t> internal_progs_comdata;
+using comdata_t = std::vector<uint8_t>;
+static std::vector<comdata_t> internal_progs_comdata;
 static std::vector<PROGRAMS_Main *> internal_progs;
 
 void PROGRAMS_MakeFile(const char *name, PROGRAMS_Main *main)
 {
-	com_data_t comdata = std::make_unique<uint8_t[]>(com_data_size);
-	assert(comdata);
-
-	memcpy(comdata.get(), &exe_block, sizeof(exe_block));
-	comdata[callback_pos] = static_cast<uint8_t>(call_program & 0xff);
-	comdata[callback_pos + 1] = static_cast<uint8_t>((call_program >> 8) & 0xff);
+	comdata_t comdata(exe_block.begin(), exe_block.end());
+	comdata.at(callback_pos) = static_cast<uint8_t>(call_program & 0xff);
+	comdata.at(callback_pos + 1) = static_cast<uint8_t>((call_program >> 8) & 0xff);
 
 	// Save the current program's vector index in its COM data
-	assert(internal_progs.size() <= UINT8_MAX);
-	const auto index = static_cast<uint8_t>(internal_progs.size());
-	comdata[sizeof(exe_block)] = index;
+	const auto index = internal_progs.size();
+	assert(index <= UINT8_MAX); // saving the index into an 8-bit space
```

Comment:
`UINT8_MAX` -> `std::numeric_limits<uint8_t>::max()`?

Reply:
This came up in some early reviews, and @dreamer's take was that this is less readable - and I agree.

It's a bloated eyeful to get the same number from the compiler. 

However I fully agree this is useful whenever the type isn't clear (such as using a typedef), or when using templates.